packages/framework/src/Migrations/Version20240131163656.php

```diff
@@ -0,0 +1,68 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Shopsys\FrameworkBundle\Migrations;
+
+use Doctrine\DBAL\Schema\Schema;
+use Shopsys\MigrationBundle\Component\Doctrine\Migrations\AbstractMigration;
+
+class Version20240131163656 extends AbstractMigration
+{
+    /**
+     * @param \Doctrine\DBAL\Schema\Schema $schema
+     */
+    public function up(Schema $schema): void
+    {
+        $this->sql('ALTER TABLE store_opening_hours ADD opening_time VARCHAR(5) DEFAULT NULL');
+        $this->sql('ALTER TABLE store_opening_hours ADD closing_time VARCHAR(5) DEFAULT NULL');
+
+        $openingHoursData = $this->sql('SELECT store_id, day_of_week, first_opening_time, first_closing_time, second_opening_time, second_closing_time FROM store_opening_hours')->fetchAllAssociative();
+
+        foreach ($openingHoursData as $openingHoursRow) {
+            $firstOpeningTime = $openingHoursRow['first_opening_time'];
+            $firstClosingTime = $openingHoursRow['first_closing_time'];
+            $secondOpeningTime = $openingHoursRow['second_opening_time'];
+            $secondClosingTime = $openingHoursRow['second_closing_time'];
+            $storeId = $openingHoursRow['store_id'];
+            $dayOfWeek = $openingHoursRow['day_of_week'];
+
+            if ($firstOpeningTime !== null && $firstClosingTime !== null) {
+                $this->insertIntoOpeningHours($firstOpeningTime, $firstClosingTime, $storeId, $dayOfWeek);
+            }
+
+            if ($secondOpeningTime !== null && $secondClosingTime !== null) {
+                $this->insertIntoOpeningHours($secondOpeningTime, $secondClosingTime, $storeId, $dayOfWeek);
+            }
+        }
+        $this->sql('DELETE FROM store_opening_hours WHERE opening_time IS NULL AND closing_time IS NULL');
```

Comment:
this may be null at any point?

Reply:
:ok_hand: 764c4cda64

Reply:
yeah, there might be entries entry with NULL opening hours for a day, see the screens below:
![image](https://github.com/shopsys/shopsys/assets/10401898/271a2007-25be-46d3-b32d-67617d0aa2f1)
![image](https://github.com/shopsys/shopsys/assets/10401898/402102f1-bcbe-4a01-91f4-cf716a13c9db)

So after the migration, you end up with a row containing null values, however, now I am not sure why this was a problem :smile: I will test the migration one more time.